### ISSUE TITLE
empty directory.files loops forever?

### ISSUE THREAD

**Issue by icco** [fog/fog-google#165]
Hey @Temikus, can you try to reproduce this?

```
     @connection = Fog::Storage::GoogleJSON.new({
       google_project: "xxxx",
       google_json_key_string: ENV["GOOGLE_JSON_KEY"],
     })
     @bucket = @connection.directories.get("xxx")
     p @bucket.files.count
``` 

I've tried with the bucket being empty and with one file.

For me this causes an infinite loop. Does this happen to you as well?

**Comment by Temikus** [comment]
@icco Did a quick check, yeah, same thing. I'll try to sort that out ASAP, though it's late here so probably need to debug in the morning.

Here's the relevant chunk of the loop:
```

From: /Users/temikus/.rbenv/versions/2.1.7/lib/ruby/gems/2.1.0/gems/fog-core-1.43.0/lib/fog/core/attributes/default.rb @ line 53 Fog::Storage::GoogleJSON::Files#max_results:

    52: model.class_eval <<-EOS, __FILE__, __LINE__
 => 53:   def #{name}
    54:     return attributes[:#{name}] unless attributes[:#{name}].nil?
    55:     if !attributes.key?(:#{name}) && !self.class.default_values[:#{name}].nil? && !persisted?
    56:       return self.class.default_values[:#{name}]
    57:     end
    58:     attributes[:#{name}]
    59:   end
    60: EOS

[2] pry(#<Fog::Storage::GoogleJSON::Files>)> step

From: /Users/temikus/.rbenv/versions/2.1.7/lib/ruby/gems/2.1.0/gems/fog-core-1.43.0/lib/fog/core/attributes.rb @ line 69 Fog::Attributes::InstanceMethods#attributes:

    68: def attributes
 => 69:   @attributes ||= {}
    70: end

[2] pry(#<Fog::Storage::GoogleJSON::Files>)> step

From: /Users/temikus/.rbenv/versions/2.1.7/lib/ruby/gems/2.1.0/gems/fog-core-1.43.0/lib/fog/core/attributes/default.rb @ line 54 Fog::Storage::GoogleJSON::Files#max_results:

    52: model.class_eval <<-EOS, __FILE__, __LINE__
    53:   def #{name}
 => 54:     return attributes[:#{name}] unless attributes[:#{name}].nil?
    55:     if !attributes.key?(:#{name}) && !self.class.default_values[:#{name}].nil? && !persisted?
    56:       return self.class.default_values[:#{name}]
    57:     end
    58:     attributes[:#{name}]
    59:   end
    60: EOS

[2] pry(#<Fog::Storage::GoogleJSON::Files>)> step

From: /Users/temikus/.rbenv/versions/2.1.7/lib/ruby/gems/2.1.0/gems/fog-core-1.43.0/lib/fog/core/attributes.rb @ line 69 Fog::Attributes::InstanceMethods#attributes:

    68: def attributes
 => 69:   @attributes ||= {}
    70: end
```

**Comment by Temikus** [comment]
The loop is in the `load(parent.files.map(&:attributes))` in `files.rb` here:
```
          if parent
            merge_attributes(parent.files.attributes)
            load(parent.files.map(&:attributes))
          end
```
, the method is just copy-pasted from GoogleXML and that logic is ancient (and working for XML), so I suspect it either never worked in the first place ( :/ ) or something's changed with JSON Storage API.

@icco - did that method ever work for you before?

**Comment by Temikus** [comment]
Another weirdness is that the XML api call goes through `Directory.files`, so maybe it's something in autoload logic...

**Comment by icco** [comment]
@Temikus nope, I had never run it before.

**Comment by Temikus** [comment]
@icco Ack, I'll dig into this. Thanks for reporting!

**Comment by Temikus** [comment]
Fixed by #167, just verified.
